### PR TITLE
ci: move CentOS 7 tests to Cirrus CI

### PR DESCRIPTION
**Pull request by adrianreber**
On CI systems using newer kernels with clone3() and old user space which does PID caching (like CentOS 7) the getpid() call returns the wrong PID as it does not know how clone3() assigns PIDs.

Avoiding glibc and doing the getpid() syscall directly fixes PID caching errors on the combination of new kernel and very old user space.

This tries to fix one of the errors described in #1503

### CODE REVIEW

**Comment by adrianreber** [comment]
@avagin: This tries to fix the clone3() error which can be seen on CentOS7. But once the getpid() returns the right value the tests are failing with time namespace problems:
```
(00.000416) Forking task with 52 pid (flags 0x0)
userns is supported
userns is supported
(00.000421) Creating process using clone3()
(00.000532) PID: real 52 virt 52
(00.000775)     52: timens: monotonic -1 989797667
(00.000803)     52: Error (criu/timens.c:99): Unable to set a monotonic clock offset: Permission denied
(00.000846) Error (criu/cr-restore.c:2521): Restoring FAILED.
(00.000963) Error (criu/cr-restore.c:1537): 52 killed by signal 9: Killed
```
I can reproduce this error locally running a CentOS 7 container on a 5.11 kernel. Any ideas how solve this error?

**Comment by adrianreber** [comment]
Thanks to @avagin's suggestion I switched all `getpid()` calls to avoid the glibc and locally the `timens` problems disappeared.

Let's see if CI also likes it.

**Comment by adrianreber** [comment]
This PR is now completely changed. Instead of changing code I moved the CentOS 7 based CI from GitHub to Cirrus CI.

On Cirrus CI we can on a CentOS 7 VM and not like on GitHub on 5.8 kernel CentOS user space which did not really work and there is no need to support such an unusual combination.

**Comment by adrianreber** [comment]
@checkpoint-restore/maintainers PTAL

**Comment by adrianreber** [comment]
If there are no objections I would like to get this merged to remove the failing CentOS 7 test from our CI runs.

File: scripts/ci/run-ci-tests.sh [hunk 1]
```diff
@@ -237,7 +218,7 @@ bash -x ./test/jenkins/criu-inhfd.sh
 if [ -z "$SKIP_EXT_DEV_TEST" ]; then
 	make -C test/others/mnt-ext-dev/ run
 fi
-#make -C test/others/exec/ run
```
Comment (Snorch) [inline scripts/ci/run-ci-tests.sh:240]: Heh it looks like `make -C test/others/exec/ run` is broken long ago...
```
+ ../../../criu/criu exec -t 186212 fake_syscall
...
Error (criu/crtools.c:162): excessive parameter for command exec
```
As criu exec is already deprecated.

**Review by Snorch — APPROVED** [review]
I'm not too good at CI. Though looks good to me.

File: scripts/ci/run-ci-tests.sh (hunk 1 above)
Comment (adrianreber) [reply scripts/ci/run-ci-tests.sh:240]: Good point. Maybe we should just remove that directory.

Comment (adrianreber) [reply scripts/ci/run-ci-tests.sh:240]: Let me open a PR to remove that directory.

**Pull request merged**